File: 2083-three-divisors/2083-three-divisors.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isThree = function(n) {
    const sqr = Math.sqrt(n);

    if (sqr % 1 != 0) {
        return false; 
    }

    function isPrime(num) {
        if (num <= 1) return false;
        for (let i = 2; i <= Math.sqrt(num); i++) {
            if (num % i === 0) {
                return false; 
            }
        }
        return true; 
    }

    return isPrime(sqr); 

};

File: 2083-three-divisors/2083-three-divisors.cs
public class Solution {
      public bool IsThree(int n) {
        double sqr = Math.Sqrt(n);

        if (sqr != (int)sqr) {
            return false;
        }

        return IsPrime((int)sqr);
    }

    private bool IsPrime(int num) {
        if (num <= 1) return false;

        for (int i = 2; i <= Math.Sqrt(num); i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }
}